pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -215,28 +236,39 @@ func (d *DualWriterMode2) DeleteCollection(ctx context.Context, deleteValidation
 	}
 
 	startStorage := time.Now()
-	res, err := d.Storage.DeleteCollection(ctx, deleteValidation, options, listOptions)
+	deletedStorage, err := d.Storage.DeleteCollection(ctx, deleteValidation, options, listOptions)
 	if err != nil {
-		log.WithValues("deleted", res).Error(err, "failed to delete collection successfully from Storage")
+		log.WithValues("deleted", deletedStorage).Error(err, "failed to delete collection successfully from Storage")
 		d.recordStorageDuration(true, mode2Str, d.resource, method, startStorage)
-		return res, err
+		return deletedStorage, err
 	}
 	d.recordStorageDuration(false, mode2Str, d.resource, method, startStorage)
 
-	areEqual := Compare(res, deleted)
-	d.recordOutcome(mode2Str, getName(res), areEqual, method)
+	areEqual := Compare(deletedStorage, deletedLegacy)
+	d.recordOutcome(mode2Str, getName(deletedStorage), areEqual, method)
 	if !areEqual {
 		log.Info("object from legacy and storage are not equal")
 	}
 
-	return res, err
+	return deletedLegacy, err
 }
 
 func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
 	var method = "delete"
 	log := d.Log.WithValues("name", name, "method", method)
 	ctx = klog.NewContext(ctx, log)
 
+	startStorage := time.Now()
```

Comment:
Why are we deleting first on Unified ?

Reply:
We could argue it the other way also, if we delete from unified but the deletion from storage fails then it'll get recreated in unified if it's edited. If we delete from storage first but deletion fails in unified then the entry in unified will be orphaned and will need to be cleaned up before we go to mode 3